### ISSUE TITLE
Can't get repeated or map field values with reflection in 2.8

### ISSUE THREAD

**Issue by adamcrume** [stepancheg/rust-protobuf#443]
There doesn't seem to be any way to get the value of a repeated field or map field using reflection in version 2.8.0.  `ReflectFieldRef::Repeated` and `ReflectFieldRef::Map` refer to `ReflectRepeated` and `ReflectMap`, but neither of those types are visible.

**Comment by stepancheg** [comment]
Indeed. Thank you for the pointer, I'll try to fix it.

**Comment by stepancheg** [comment]
Full proper reflection implementation is merged in v2.9 branch.

Feedback is very welcome.

There's one decision I need to make before releasing that version.

There's `ReflectValueRef` type. It is:

```
pub enum ReflectValueRef<'a> {
    ...
    Enum(&'static EnumValueDescriptor), # in stable
    Enum(&'static EnumDescriptor, i32), # in master
}
```

Master version is better compatible with an option to store unknown enum values.

Changing to master will be backwards incompatible. The question is how much damage is in changing API.

**Comment by stepancheg** [comment]
Released in 2.9.0.

**Comment by adamcrume** [comment]
What's the status of this?  It looks like version 2.9.0 was released and then yanked.
